### DISCUSSION TITLE
Doctrine uses too much RAM

### DISCUSSION THREAD

**Discussion by koredalin** [symfony/symfony#42402] [Q&A] [answered]
Hello All,
I have an issue with Doctrine.
I have assembled a framework with different components without Symfony framework, but with Doctrine and Symfony cache.
The issue is that Doctrine allocates too much memory when I make a simple "find(id)" query thru the object repository.
I have searched many pages in the internet.
I have stopped Doctrine SQLLogger. It helped at first. But after I have tried a cache warmup for a second time - the memory leak started again.
I have found how to set the proxy files generation. But it was too late.

![image](https://user-images.githubusercontent.com/6977867/128490489-e1fb7bf5-2872-4476-ad68-fe7d343d5fea.png)

Please, give me an advice.
Thank you
Hristo (edited)

**Comment by koredalin** [comment]
Also - if I make persist(); flush(); of one row into the database.. - The record is made into the database. But there is no returned object. Nothing is returned.

**Comment by dbrumann** [comment, marked as answer]
The doctrine project has their own discussions: https://github.com/doctrine/orm/discussions. You might have better luck there.

That being said, you might want to look at clearing the ObjectManager regularly. It's hard to tell without seeing any code, but I find that often times old entity manager state can not be collected by the garbage collector, which can cause the memory issues that you describe.

**Reply by dbrumann** [reply]
You might also want to look into using a profiler like blackfire.io or the https://github.com/BitOne/php-meminfo extension to see what is eating up memory and where. That will help you see where you need to make adjustments or you can provide the Doctrine team with more details, where the memory leaks happen if they are inside Doctrine itself.

**Reply by koredalin** [reply]
Thank you, @dbrumann,
I'll debug the program tomorrow and I'll write the result here and eventually in Doctrine discussions.

**Reply by ruudk** [reply]
I highly recommend https://github.com/NoiseByNorthwest/php-spx. Great free profiler that's easy to use locally.

**Reply by koredalin** [reply]
@ruudk,
Thank you for the idea.

**Comment by koredalin** [comment]
Hello All,
I found the issue.
print_r() allocates too much memory because it makes many recursive operations. var_export() works fine.
My repository return the expected object. My repository service does not. I have missed the return statement there.

Thanks for the comments.

Best regards
Hristo
